### PR TITLE
Onboarding - Login and Signup v2

### PR DESCRIPTION
**Pull request by MGLarsenSW**
### Description
This pull request is the complete backend functionality of the relevant PBIs of the Onboarding team.

### Changes
- Added new routing for Login and Signup post requests: they are now handled in credentialsRoutes.js.
- Changed the models of the ContencCreators to match the newer specifications, i.e. removed "motivation" and made "firstname" and "lastname" one field.
- Added token creation upon receiving a login POST request in credentialsRouts.js, which the frontend is able to make use of,
- Added hashing of password, which are handled in credentialsRoutes.js.
- Added error messages that the frontend login page can use when invalid credentials are used, also in credentialsRouts.js.

Additionally, there are no new dependencies.

### Related Issues
There are no related issues to the above functionalities.

### Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x]  Code follows the established coding style and guidelines of the project.
- [x]  All new and existing tests related to the changes have passed.
- [x]  Any necessary dependencies or new packages have been properly documented.
- [x]  Pull request title and description are clear and descriptive.
- [x]  Reviewers have been assigned to the pull request.
- [x]  Any potential security implications have been considered and addressed.
- [x]  Performance impact of the changes has been evaluated, if relevant.

### CODE REVIEW

**Comment by sofiiagran** [comment]
It does not pass all the tests...

File: __tests__/routes/contentCreator.test.js [hunk 1]
```diff
@@ -30,19 +29,19 @@ describe('DELETE /api/creators/:id', () => {
 
   beforeEach(async () => {
     // Insert the fake creator into the database
-    await db.collection('contentcreators').insertOne(fakeCreator);
+    await db.collection('content-creators').insertOne(fakeCreator);
```
[the remark this answers is not in the document]
Comment (sofiiagran) [reply __tests__/routes/contentCreator.test.js:32]: I am also a little curious on why it is changed to content-creators, but it is not breaking and the code works

**Review by sofiiagran — APPROVED** [review]
I think it looks good, I am not sure about the kebab-case on database name, but it is not critical and the code works fine on my computer. However, I would like another person to approve before you merge.

File: __tests__/fixtures/fakeContentCreator.js [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
-    name: "test1 test1",
-    approved: true
-  }
-}
\ No newline at end of file
+module.exports = function makeFakeContentCreator(email = 'fake@gmail.com') {
+    return {
+        email: email,
+        password: encrypt('ABC123456!'),
+        name: 'Fake name',
```
Comment (MortenMunk) [inline __tests__/fixtures/fakeContentCreator.js:8]: Does this model still only use name, or has this not been changed to firstname and lastname yet?

File: __tests__/routes/contentCreator.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (MortenMunk) [reply __tests__/routes/contentCreator.test.js:32]: Maybe PR requester can answer why this is written as kebab case? @MGLarsenSW

File: __tests__/setup/globalConfigMongo 2.json
Comment (MortenMunk) [inline __tests__/setup/globalConfigMongo 2.json:1]: I believe this can be deleted?

File: __tests__/setup/globalConfigMongo.json
Comment (MortenMunk) [inline __tests__/setup/globalConfigMongo.json:1]: I believe this can be deleted?

File: applications/content-creator-applications/domain/contentCreatorApplication.js
Comment (MortenMunk) [inline applications/content-creator-applications/domain/contentCreatorApplication.js:1]: Why is this changed from firstname and lastname to just name?
Shouldn't it be firstname and lastname? or am i mistaken?

File: models/ContentCreatorApplication.js
Comment (MortenMunk) [inline models/ContentCreatorApplication.js:1]: Same, why are we not using firstname and lastname?

**Review by MortenMunk — CHANGES_REQUESTED** [review]
I have a few questions regarding the changes, and some files I believe are unnecessary.
Other than that, I think it is fine 👍

**Review by JackSMunn — COMMENTED** [review]
Good job! Code looks fine.

See other reviewers comments, I have no additions.

File: __tests__/routes/contentCreator.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (MGLarsenSW) [reply __tests__/routes/contentCreator.test.js:32]: it was changed to kebab case since it needs to be case sensitive and DB does not accept camelCase when i did the tests. the reason @sofiiagran yours is not breaking is because i change your test to match. also i think we created a file called the exact same so it tried to merged ours together. i have updated your files to match ours since yours was a hardcoded mail called test1@mail.dk, instead of the one i have created called testmail@gmail.com that can be updated in the tests and now it also matches the same email address as all the other tests created by other teams.

File: __tests__/fixtures/fakeContentCreator.js (hunk 2 above)
Comment (MGLarsenSW) [reply __tests__/fixtures/fakeContentCreator.js:8]: it only uses name for now. we will update this file to contain first name and last name in a future sprint, but for now we only use name

File: models/ContentCreatorApplication.js
Comment (MGLarsenSW) [reply models/ContentCreatorApplication.js:1]: for now the frontend is setup with only a field for name, so we dont need first name and last name, just a name. this will be changed in a future sprint

**Review by MortenMunk — APPROVED** [review]

**Comment by MortenMunk** [comment]
I have approved for now, but we should consider changing from name, to firstName and lastName like in the mobile app... This is for a future sprint tho, as PR requester explained.
